libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1186,11 +1186,11 @@ string ABIFunctions::abiDecodingFunctionArrayAvailableLength(ArrayType const& _t
 					mstore(array, length)
 					dst := add(array, 0x20)
 				</dynamic>
-				let src := offset
-				if gt(add(src, mul(length, <stride>)), end) {
+				let srcEnd := add(offset, mul(length, <stride>))
+				if gt(srcEnd, end) {
 					<revertInvalidStride>()
```

Comment:
Actually... invalid stride? Why was the revert message here ever "invalid stride"? Am I missing something here?
This is really an out-of-bounds error, isn't it?

Reply:
What does "invalid stride" even mean as a runtime error ;-)?

Reply:
true

Reply:
`length` is already checked to be non-gigantic, right?

Reply:
Ah, indirectly through the allocation.